### DISCUSSION TITLE
Does Function calling work with any other models than OpenAi's  in Shellgpt?

### DISCUSSION THREAD

**Discussion by akhil3417** [TheR1D/shell_gpt#506] [Q&A]
would be great if other models like Mistral would be supported ,here are the docs
https://docs.mistral.ai/guides/function-calling/

**Comment by TheR1D** [comment]
Implementing all features for each LLM model could be quite time consuming. ShellGPT by default uses OpenAI models and all features will be primarely implemented using GPT models from OpenAI. I believe ShellGPT might have forks that will be optimized for other LLMs.
